projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6486,21 +6485,21 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
      */
     endRowTransaction(commit: boolean, row: IgxRow) {
         row.newData = this.transactions.getAggregatedValue(row.id, true);
-        const rowArgs = row.createEditEventArgs();
 
+        let rowEditArgs;
         if (!commit) {
-            this.rowEditExit.emit(rowArgs);
             this.transactions.endPending(false);
         } else {
-            const rowEditArgs = this.gridAPI.update_row(row, row.newData);
+            rowEditArgs = this.gridAPI.update_row(row, row.newData);
             if (rowEditArgs?.cancel) {
                 return true;
             }
-
-            this.rowEditExit.emit(rowArgs);
         }
 
         this.crudService.endRowEdit();
+
+        const nonCancelableArgs = row.createPostCommitEditEventArgs({...row.data, ...rowEditArgs});
```

Comment:
Merging the row data with the edit args will produce some really weird old values here - with their own `oldValue`, `newValue`, etc inside :)